integration-test/src/test/java/org/apache/iotdb/pipe/it/autocreate/IoTDBPipeAlterIT.java

```diff
@@ -99,6 +100,71 @@ public void testBasicAlterPipe() throws Exception {
       Assert.assertEquals("STOPPED", showPipeResult.get(0).state);
     }
 
+    // Alter pipe (modify)
+    try (final Connection connection = senderEnv.getConnection();
+        final Statement statement = connection.createStatement()) {
+      statement.execute("alter pipe a2b modify source ('source.path'='root.timecho')");
+    } catch (SQLException e) {
+      fail(e.getMessage());
+    }
+
+    // Show pipe
+    try (final SyncConfigNodeIServiceClient client =
+        (SyncConfigNodeIServiceClient) senderEnv.getLeaderConfigNodeConnection()) {
+      final List<TShowPipeInfo> showPipeResult = client.showPipe(new TShowPipeReq()).pipeInfoList;
+      Assert.assertEquals(1, showPipeResult.size());
+      // Check status
+      Assert.assertEquals("STOPPED", showPipeResult.get(0).state);
+      // Check configurations
+      Assert.assertTrue(showPipeResult.get(0).pipeExtractor.contains("source.path=root.timecho"));
+      Assert.assertTrue(
+          showPipeResult.get(0).pipeProcessor.contains("processor=do-nothing-processor"));
+      Assert.assertTrue(
+          showPipeResult
+              .get(0)
+              .pipeConnector
+              .contains(String.format("node-urls=%s", receiverDataNode.getIpAndPortString())));
+      // Check creation time and record last creation time
+      Assert.assertTrue(showPipeResult.get(0).creationTime > lastCreationTime);
+      lastCreationTime = showPipeResult.get(0).creationTime;
+      // Check exception message
+      Assert.assertEquals("", showPipeResult.get(0).exceptionMessage);
+    }
+
+    // Alter pipe (replace)
+    try (final Connection connection = senderEnv.getConnection();
+        final Statement statement = connection.createStatement()) {
+      statement.execute(
+          "alter pipe a2b replace source ('source' = 'iotdb-source','source.path'='root.timecho.wf01')");
+    } catch (SQLException e) {
+      fail(e.getMessage());
+    }
+
+    // Show pipe
+    try (final SyncConfigNodeIServiceClient client =
+        (SyncConfigNodeIServiceClient) senderEnv.getLeaderConfigNodeConnection()) {
+      final List<TShowPipeInfo> showPipeResult = client.showPipe(new TShowPipeReq()).pipeInfoList;
+      Assert.assertEquals(1, showPipeResult.size());
+      // check status
+      Assert.assertEquals("STOPPED", showPipeResult.get(0).state);
+      // check configurations
+      Assert.assertTrue(showPipeResult.get(0).pipeExtractor.contains("source=iotdb-source"));
+      Assert.assertTrue(
+          showPipeResult.get(0).pipeExtractor.contains("source.path=root.timecho.wf01"));
```

Comment:
Ensure that the show pipe result does not contain the legacy "source.pattern" parameter.

Reply:
I did not add the "source.pattern" parameter. :-)